cpp/ql/test/query-tests/Critical/NewFree/test.cpp

```diff
@@ -371,3 +371,56 @@ void test12(bool cond)
 		free(z); // GOOD
 	}
 }
+
+// ---
+
+class MyBuffer13
+{
+public:
+	MyBuffer13(int size)
+	{
+		buffer = (char *)malloc(size * sizeof(char));
+	}
+
+	~MyBuffer13()
+	{
+		free(buffer); // GOOD
+	}
+
+	char *getBuffer() // note: this should not be considered an allocation function
+	{
+		return buffer;
+	}
+
+private:
+	char *buffer;
+};
+
+class MyPointer13
+{
+public:
+	MyPointer13(char *_pointer) : pointer(_pointer)
+	{
+	}
+
+	MyPointer13(MyBuffer13 &buffer) : pointer(buffer.getBuffer())
+	{
+	}
+
+	char *getPointer() // note: this should not be considered an allocation function
+	{
+		return pointer;
+	}
+
+private:
+	char *pointer;
+};
+
+void test13()
+{
+	MyBuffer13 myBuffer(100);
+	MyPointer13 myPointer2(myBuffer);
+	MyPointer13 myPointer3(new char[100]);
+
+	delete myPointer2.getPointer(); // GOOD
```

Comment:
I must be missing something here. I'd expect `myPointer2.getPointer()` to return `pointer` from `myPointer2`, which is set in the constructor call `myPointer2(myBuffer)`, and `myBuffer` uses `malloc`. So why is this `// GOOD`?

Reply:
The test was supposed to delete `myPointer3`, sorry (the point being that the query can't tell the difference, so has to be conservative in this case).  Updated.